packages/redux-toolbelt-thunk/src/makeThunkAsyncActionCreator.d.ts

```diff
@@ -39,14 +39,6 @@ export declare function makeThunkAsyncActionCreator<A extends Action, P = any>(
   options: makeThunkAsyncActionCreator.Options<A>
 ): ThunkAsyncActionCreator<A>
 
-export interface ActionCreatorProps {
-  /**
-   * Exposes TYPE as static member.
-   */
-  Type: string;
-  toString: () => string;
-}
-
 export type ActionCreator<A extends Action> = (
   (payload?: any, meta?: any) => GenericAction<A['payload'], A['meta']>
```

Comment:
```suggestion
  (payload?: A['payload'], meta?: A['meta']) => GenericAction<A['payload'], A['meta']>
```

I feel that this should be fixed in the base `ActionCreator` definition in `redux-toolbelt`.

Reply:
I agree that it should be on base, however limiting the function parameters misses the goal. The function will return the typed action, but the parameters should stay type loose I think

Reply:
What goal exactly does it miss?
The point of types is to prevent mistakes, when you keep it any - you leave the problem as is.

If the single type I purposed is not enough, than it should be extended to whatever types can be accepted.

Reply:
I agree with what you saying, it should be typed. I just thought that they will maybe enter some input to the action creator to later get another output on the Action.
How about this code:

export type ActionCreator<O extends Action, I extends O> = (
  (payload?: I['payload'], meta?: I['meta']) => GenericAction<O['payload'], O['meta']>
)

They will be able to define input and output, but if only input was defined, it will act as output
?

Reply:
Sorry for not understanding, I probably don't remember the API anymore,
But at what use-case does this happen? That the payload of the output `action` is different from what is passed to the `action creator`?